### PR TITLE
GPC-500: Add sonar token

### PR DESCRIPTION
**Pull request by oskarwilliams**

### CODE REVIEW

**Comment by olevett** [comment]
This in the build logs looks interesting... 
image

File: .github/workflows/sam-pull-request.yml [hunk 1]
```diff
@@ -16,28 +16,39 @@ jobs:
...
+        with:
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
+
+      - name: Build and unit tests
+        run: ./gradlew clean build
+
+      - name: Perform Static Analysis
+        if: ${{ github.actor != 'dependabot[bot]' }}
```
Comment (olevett) [inline .github/workflows/sam-pull-request.yml:47]: So we don't do this on dependabot? This feels...odd, but presumably there's a reason?

File: build.gradle [hunk 2]
```diff
@@ -1,3 +1,76 @@
...
+                //the jacoco extension may be disabled for some projects
+                if (testTask.extensions.getByType(JacocoTaskExtension).isEnabled()) {
+                    sourceSets subproject.sourceSets.main
+                    executionData(testTask)
+                } else {
+                    logger.warn('Jacoco extension is disabled for test task \'{}\' in project \'{}\'. this test task will be excluded from jacoco report.',testTask.getName(),subproject.getName())
+                }
+            }
+
+            // To automatically run `test` every time `./gradlew codeCoverageReport` is called,
```
Comment (olevett) [inline build.gradle:60]: This comment feels a bit redundant given we've got the code uncommented

File: .github/workflows/sam-pull-request.yml (hunk 1 above)
Comment (oskarwilliams) [reply .github/workflows/sam-pull-request.yml:47]: Nope this was just while I tested, removed

File: build.gradle (hunk 2 above)
Comment (oskarwilliams) [reply build.gradle:60]: Removed

File: .github/workflows/sam-pull-request.yml
Comment (olevett) [inline .github/workflows/sam-pull-request.yml:1]: Presumably we're calling this "sam" to seperate it from the ECS/Terraform infra?

**Review by olevett — APPROVED** [review]

File: .github/workflows/sam-pull-request.yml
Comment (oskarwilliams) [reply .github/workflows/sam-pull-request.yml:1]: Yep

**Pull request merged**